src/sentry/identity/vsts/provider.py

```diff
@@ -68,18 +78,59 @@ def get_pipeline_views(self):
     def get_refresh_token_headers(self):
         return {"Content-Type": "application/x-www-form-urlencoded", "Content-Length": "1654"}
 
-    def get_refresh_token_params(self, refresh_token, *args, **kwargs):
+    def get_refresh_token_params(self, refresh_token, identity, *args, **kwargs):
+
+        client_secret = options.get("vsts.client-secret")
+
+        # The token refresh flow does not operate within a pipeline in the same way
+        # that installation does, this means that we have to use the identity.scopes
+        # to determine which client_secret to use.
+        #
+        # If "vso.code" is missing from the identity.scopes, we know that we installed
+        # using the "vsts-limited.client-secret" and therefore should use that to refresh
+        # the token.
+        if "vso.code" not in identity.scopes:
+            client_secret = options.get("vsts-limited.client-secret")
+
         oauth_redirect_url = kwargs.get("redirect_url")
         if oauth_redirect_url is None:
             raise ValueError("VSTS requires oauth redirect url when refreshing identity")
         return {
             "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
-            "client_assertion": self.get_oauth_client_secret(),
+            "client_assertion": client_secret,
             "grant_type": "refresh_token",
             "assertion": refresh_token,
             "redirect_uri": absolute_uri(oauth_redirect_url),
         }
 
+    def refresh_identity(self, identity, *args, **kwargs):
+        """
+            Almost identical to OAuth2Provider but passes through the identity
+            into `get_refresh_token_params`. We do this because the identity.scopes
+            tell us which client_secret to use.
+        """
+        refresh_token = identity.data.get("refresh_token")
+
+        if not refresh_token:
+            raise IdentityNotValid("Missing refresh token")
+
+        data = self.get_refresh_token_params(refresh_token, identity, *args, **kwargs)
```

Comment:
@MeredithAnya Could we avoid having to provide a new definition of `refresh_identity` if we just applied `identity` as a `kwarg` when calling `get_refresh_token_params` in the base class?